### PR TITLE
Update Dockerfile to address ssh-keygen errors on current image v2

### PR DESCRIPTION
**Pull request by lindsleycj**
I took it as a challenge to come up with a generic way to implement the changes needed to resolve issue #162 and proposed in PR #163

### CODE REVIEW

**Review by mikedorfman — APPROVED** [review]
Looks good to me!

File: Dockerfile [hunk 1]
```diff
@@ -41,6 +41,11 @@ RUN \
         curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/linux_64bit/session-manager-plugin.rpm" -o "session-manager-plugin.rpm" &&\
         yum install -y session-manager-plugin.rpm
 
+# Add user to /etc/passwd for keygen in Makefile
+ARG USER
+RUN \
+        echo "user:x:${USER}:0:root:/:/bin/bash" >> /etc/passwd
```
Comment (reweeden) [inline Dockerfile:47]: Maybe use the `useradd` command instead? That would feel less error prone to me.

**Review by reweeden — COMMENTED** [review]
So I guess it's happening because we run the docker command with 
```
--user `id -u`
```
which we need to do so that file permissions map correctly to the host's file system.

File: Dockerfile (hunk 1 above)
Comment (lindsleycj) [reply Dockerfile]: @reweeden I've been playing with using `useradd` and it keeps hanging at the `exporting layers` part of the image build.  Have you seen this before?  

```
RUN \
         useradd -M -u ${USER} user
```

**Pull request merged**